tests/test_Source/test_supernovae_pop.py

```diff
@@ -0,0 +1,76 @@
+from slsim.Sources.Supernovae.supernovae_pop import (
+	calculate_star_formation_rate,
+	delay_time_distribution,
+)
+from slsim.Sources.Supernovae.supernovae_pop import SNIaRate
+from astropy.cosmology import FlatLambdaCDM
+import numpy as np
+import numpy.testing as npt
+
+
+def test_calculate_star_formation_rate():
+	z = 0
+	npt.assert_almost_equal(calculate_star_formation_rate(z), 0.0118, decimal=4)
+	z = 3
+	npt.assert_almost_equal(calculate_star_formation_rate(z), 0.1565, decimal=3)
+
+
+def test_delay_time_distribution():
+	t_d = -1
+	npt.assert_equal(delay_time_distribution(t_d), np.nan)
+	t_d = 5
+	npt.assert_almost_equal(delay_time_distribution(t_d), 5 ** (-1.08), decimal=4)
+
+
+class TestSNIaRate():
+    def setup_method(self):
+        self.cosmo = FlatLambdaCDM(H0=70, Om0=0.3)
+        self.z_max = 10
+        self.sne_rate = SNIaRate(
+        	cosmo=self.cosmo,
+        	z_max=self.z_max,
+        )  
+
+    def test_z_from_time(self):
+    	z_true = 0
+		t = self.cosmo.age(z_true)
```

Comment:
odd formatting, not sure this works

Reply:
VSCode was making me revert to an old version, so I copied the code before I did and the spacing got messed up. I fixed it so everything should work now!